### DISCUSSION TITLE
A way to load custom module (library) into another module?

### DISCUSSION THREAD

**Discussion by manojdhrma** [nWidart/laravel-modules#1501] [General]
Hello, 

I am very interested with this laravel-modules. It is awesome to work with and scaling the application with modules very easy.
I'm trying to experiment with different approaches with this. 

One idea is that, I need to load a custom module as a library component into one of the main module to function inside as an instance. The idea actually works like a plugin. And I can set it to enable or disable from admin area or similar place. 
(I am using "laravel-modules-livewire" to have custom modules)

Example use:, 
I have created a module called "Dashboard" and I'm trying to load a custom module created in 'libraries' folder called "Chat" into the Dashboard Module. The 'Chat' module can access using its own route but I need to prevent it from accessing as a public route and only allow it to access within "Dashboard" module. 

I have actually load the custom module using an iFrame with its route. It works as I expected. But the route is still can access publicly and need to prevent that.  

So, 

1. I need to know using an iframe is the correct way to load a module into another module or is there a better way in "laravel-modules" to achieve this ?
2. How can I prevent the custom module route from accessing publicly and only allow to access within the module I need?

**Comment by dcblogdev** [comment]
Since you're using Livewire I would embed the livewire component into the dashboard view something like:

```
<livewire::chat.log />
```

which would load a component located in a Chat module called log.

**Reply by manojdhrma** [reply]
Thank you for this. It is a good idea.
